src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -96,14 +96,66 @@ public DockerClient(@Nonnull Launcher launcher, @CheckForNull Node node, @CheckF
      * @param entrypoint The command to execute in the image container being run.
      * @return The container ID.
      */
-    public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNull String args, @CheckForNull String workdir, @Nonnull Map<String, String> volumes, @Nonnull Collection<String> volumesFromContainers, @Nonnull EnvVars containerEnv, @Nonnull String user, @Nonnull String entrypoint) throws IOException, InterruptedException {
+    public String runWithEntrypoint(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNull String args, @CheckForNull String workdir, @Nonnull Map<String, String> volumes, @Nonnull Collection<String> volumesFromContainers, @Nonnull EnvVars containerEnv, @Nonnull String user, @Nonnull String entrypoint) throws IOException, InterruptedException {
+        ArgumentListBuilder argb = constructArgs(args, workdir, volumes, volumesFromContainers, containerEnv, user);
+        argb.add("--entrypoint").add(entrypoint).add(image);
+
+        LaunchResult result = launch(launchEnv, false, null, argb);
+        if (result.getStatus() == 0) {
+            return result.getOut();
+        } else {
+            throw new IOException(String.format("Failed to run image '%s'. Error: %s", image, result.getErr()));
+        }
+    }
+
+    /**
+     * Run a docker image, passing an arbitrary command.
+     *
+     * @param launchEnv Docker client launch environment.
+     * @param image The image name.
+     * @param args Any additional arguments for the {@code docker run} command.
+     * @param workdir The working directory in the container, or {@code null} for default.
+     * @param volumes Volumes to be bound. Supply an empty list if no volumes are to be bound.
+     * @param volumesFromContainers Mounts all volumes from the given containers.
+     * @param containerEnv Environment variables to set in container.
+     * @param user The <strong>uid:gid</strong> to execute the container command as. Use {@link #whoAmI()}.
+     * @param command The command to execute in the image container being run.
+     * @return The container ID.
+     */
+    public String runWithCommand(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNull String args, @CheckForNull String workdir, @Nonnull Map<String, String> volumes, @Nonnull Collection<String> volumesFromContainers, @Nonnull EnvVars containerEnv, @Nonnull String user, @CheckForNull String... command) throws IOException, InterruptedException {
+        ArgumentListBuilder argb = constructArgs(args, workdir, volumes, volumesFromContainers, containerEnv, user);
+
+        if (command != null) {
```

Comment:
Would be simpler just to have one method with both an `entrypoint` and a `command`.